File: python/utilidades.py
# FUNCIONES DE UTILIDAD

from typing import Any, Tuple

def linea(largo: int) -> None:
  print('═' * largo)

def encabezado(titulo: str, altura: int):
    linea(altura)
    print(titulo.upper().center(altura))
    linea(altura)
    
# conversion de string

def stringToNumber(msj: str, fn: Any) -> Any: #con fn agregamos que la funcion nos puede retornar cualquier tipo de dato, eso se especifica donde se vaya a aplicar
    number: int
    while True:
      try: 
        number = fn(input(msj))
        return number
      except ValueError:
        print('Error, signo invalido, escriba un numero por favor')
  

def readString(msj: str, aviso: str) -> str:
  texto: str = ''
  while True:
    texto = input(msj + ': ')
    if texto.strip():
      return texto.strip()
    else:
      print(aviso)

File: Js/src/utilidades.js
//funciones de utilidad
import PromptSync from "prompt-sync"
const input = PromptSync()

const readLineNumber = (msj = '',) => {
  let numero = 0;
  numero = Number(input(msj + ': '));
  return numero;
};
const readLine = (msj = '',) => {
  let string = '';
  while (true) {
    string = (input(msj + ': '));
    if (string.trim() != '') {
      return string
    } else {
      console.error('no puedes dejar  el campo vacio')
      
    }
  }
};

const centrar = (texto = '', largo = 0) => {
  let relleno = (largo - texto.length) / 2
  let textoCentrado = ' '.repeat(relleno)
  console.log(textoCentrado + texto + textoCentrado)
};

const line = (largo = 0) => {
  console.log('═'.repeat(largo));
};

const encabezado = (titulo = '', linea = 0) => {
  line(linea)
  centrar(titulo, linea) //a las acciones se le colocan parentesis para que se ejecute
  line(linea)
};

const capitalizar = (texto = '') => {
  return texto.charAt(0).toUpperCase() + texto.slice(1).toLowerCase();
}


export { encabezado, line, readLineNumber, capitalizar, readLine}



/*
    
  



*/